coderd/database/queries/provisionerjobs.sql

```diff
@@ -282,6 +282,32 @@ SET
 WHERE
 	id = $1;
 
+-- name: UpdatePrebuildProvisionerJobWithCancel :many
+-- Cancels all pending provisioner jobs for prebuilt workspaces on a specific template version.
+-- This is called when a new template version is promoted to active.
+UPDATE provisioner_jobs
+SET
+	canceled_at = @now::timestamptz,
+    completed_at = @now::timestamptz
+WHERE id IN (
+    SELECT pj.id
+    FROM provisioner_jobs pj
+    INNER JOIN workspace_builds wb ON wb.job_id = pj.id
+    INNER JOIN workspaces w ON w.id = wb.workspace_id
+    WHERE
+        w.template_id = @template_id
+        AND wb.template_version_id = @template_version_id
+		-- Prebuilds system user: prebuild-related provisioner jobs
+        AND w.owner_id = 'c42fdf75-3097-471c-8c33-fb52454d81c0'::uuid
+        AND wb.transition = 'start'::workspace_transition
```

Comment:
Suggestion: only do this for the first workspace build.

```suggestion
        AND wb.transition = 'start'::workspace_transition
        AND wb.build_number = 1
```

Reply:
Addressed in https://github.com/coder/coder/pull/20387/commits/fac079e9c50978a6dbfbb897237cbd72667cca53